File: java/Strings/Day7/BeautySum.java
package Day7;
import java.util.*;

public class BeautySum {
    public int getMinCnt(int[] freq) {
        int minCnt = Integer.MAX_VALUE;

        for (int i = 0; i < 26; i++) {
            if (freq[i] != 0) {
                minCnt = Math.min(minCnt, freq[i]);
            }
        }
        return minCnt;
    }


    public int getMaxCnt(int[] freq) {
        int maxCnt = 0;

        for (int i = 0; i < 26; i++) {
            maxCnt = Math.max(maxCnt, freq[i]);
        }

        return maxCnt;
    }


    public int beautySum(String s) {
        int sum = 0;

        for (int i = 0; i < s.length(); i++) {
            int[] freq = new int[26];

            for (int j = i; j < s.length(); j++) {
                freq[s.charAt(j) - 'a']++;

                int beauty = getMaxCnt(freq) - getMinCnt(freq);

                sum += beauty;
            }
        }
        return sum;
    }


    

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter string: ");
        String s = sc.nextLine();

        BeautySum obj = new BeautySum();
        int result = obj.beautySum(s);

        System.out.println("Beauty Sum: " + result);
        sc.close();
    }
}

File: python/Strings/day7/beautySum.py
def beautySum(s: str) -> int:
    total_beauty = 0
    n = len(s)

    for i in range(n):
        freq = [0] * 26

        for j in range(i, n):
            freq[ord(s[j]) - ord('a')] += 1

            max_cnt = 0
            min_cnt = float('inf')

            for f in freq:
                if f > 0:
                    max_cnt = max(max_cnt, f)
                    min_cnt = min(min_cnt, f)

            total_beauty += (max_cnt - min_cnt)
    
    return total_beauty



# -------- main --------
if __name__ == "__main__":
    s = input("Enter string: ")
    result = beautySum(s)

    print("Beauty Sum:", result)